src/realm/object-store/sync/impl/sync_client.hpp

```diff
@@ -74,6 +74,16 @@ struct SyncClient {
                 c.pong_keepalive_timeout = config.timeouts.pong_keepalive_timeout;
             if (config.timeouts.fast_reconnect_limit > 1000)
                 c.fast_reconnect_limit = config.timeouts.fast_reconnect_limit;
+            if (config.timeouts.resumption_delay_interval >= 1000)
```

Comment:
Why do we want to make the default values the floor rather than just the default? Tbh, I've always been skeptical of trying to force these arbitrary "sensible" ceilings. I'm def for threading this value down to the client but I'd be more in favor of just using whatever the client has passed in.

Reply:
Even though it's the default value, the floor values are meant to prevent spamming the server too much. I don't think it would be reasonable to have a bunch of clients trying to reconnect every 250ms with a max retry of 2 secs... I chose 1 second and 30 seconds as a reasonable tradeoff of allowing lower values without spamming the server too much.

tbh - I should use the floor values instead of the defaults if the passed in value is lower than the floor.

Reply:
Removed most of the checks - SyncClient will now throw an InvalidArgument exception if the multiplier is less than 1 or the divisor is less than 0. Also added an info debug that will print if the resumption delay interval is less than 1 second.

Reply:
Setting a minimum value was done at the request of the (pre-acquisition) server team. Some of our original users liked to set very stupid values for every single value we let them set and this resulted in a lot of wasted support effort when everything broke due to nonsensical settings.

The current server team wants the reconnect jitter to still exist, so exposing a setting to effectively turn it off seems questionable?

Reply:
I agree with requiring at least minimum values to avoid spamming the server too much. Should we make the minimum jitter 2 so it can't be disabled - or we could just force it to always be 4 and ignore the configured value.

On the other hand, perhaps it would be better to not make these values configurable to avoid the risk of breaking sync reconnects?

Reply:
We'd actually need a *maximum* value; smaller numbers increase the jitter. I think that field in particular doesn't make much sense to be adjustable.

Reply:
Thanks - updated the `delay_jitter_divisor` value so it is no longer configurable.